packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/explore/TablesViewer.tsx

```diff
@@ -33,12 +46,47 @@ type Props = {
   isLiveQuery: boolean;
 };
 
-export function TablesViewer({ table, query, isLiveQuery }: Props) {
+export function TablesViewer({ table, isLiveQuery }: Props) {
   const { id: chainId } = useChain();
   const indexer = indexerForChainId(chainId);
-  const { data: tableData, isPending, isError, error } = useTableDataQuery({ table, query, isLiveQuery });
+  const [query, setQuery] = useQueryState("query", parseAsString.withDefault(""));
   const [globalFilter, setGlobalFilter] = useQueryState("filter", parseAsString.withDefault(""));
   const [sorting, setSorting] = useQueryState("sort", parseAsJson<SortingState>().withDefault(initialSortingState));
+  const [page, setPage] = useQueryState("page", parseAsInteger.withDefault(0));
+  const [pageSize, setPageSize] = useQueryState("pageSize", parseAsInteger.withDefault(10));
+  const { data: tableData, isPending, isFetching, isError, error } = useTableDataQuery({ table, query, isLiveQuery });
+  const isLoading = isPending || (isFetching && !isLiveQuery);
+  const [pagination, setPagination] = useState<PaginationState>({
+    pageIndex: page,
+    pageSize,
+  });
```

Comment:
why the extra state here when the state already lives in the url? why not just use query as source of truth instead of carrying it around twicE?

Reply:
Initially I didn't like the URL query string it produced but actually it probably doesn't matter that much, and doing it like that is def nicer code, adjusted.

Reply:
share more? what was the before/after? I didn't run this myself so I can't tell

Reply:
Now it looks like this `&pagination=%7B%22pageSize%22:10,%22pageIndex%22:4%7D` and before it was `&page=2&pageSize=10`. But I suppose it's a minor difference given most of the rest of the URL query is quite unreadable anyhow.

Reply:
ahh okay, I much prefer the nice/user friendly params